lib/huginn_agent.rb

```diff
@@ -1,17 +1,45 @@
 require 'huginn_agent/version'
 
 class HuginnAgent
```

Comment:
What about a Huginn namespace instead?
`Huginn::Agent`
And rely on inheritance from `Huginn::Agent` rather than naming convention for discoverability?